### PR TITLE
Make ancient DNA damage correction optional

### LINKED ISSUE #456: Split ancientDNA sub-workflow into separate sub-workflows for running damage analysis and consensus correction
### Description of feature

Currently, enabling the `ancientDNA` workflow automatically enables the correction of the consensus sequence reported by the *de novo* assembler and the analysis of the presence of ancient DNA damage. The first step is mainly driven by the issue that we observed that MEGAHIT generates contig sequences that are riddled with nonsense mutations or stop codons when ancient DNA damage is present but were not supported by the short-read data.

When assembling with metaSPAdes, we do not observe such a behaviour and therefore the consensus correction might not be necessary. However, we still would like to run the evaluation of the amount of aDNA damage present using pyDamage.

It would be very helpful, if the two modules in this sub-workflow could be separated from each other, so they can be switched on and off separately.

### PR DESCRIPTION
**Pull request by jfy133**
This makes the damage correction subworkflow optional, i.e. ytou can run just pyDamage, and optionally turn on damage correction using freebayes and bcftools.

@alexhbnr please can you test, and let me know if you would refer to have it opt in (rather than opt out).

Also note this _does not yet_ include the hotfix in the patch for #449 , I will add that later as it requires more thinking due to the more complex inputs to bin refinement since domain classification has come in.

fixes #456

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by alexhbnr — CHANGES_REQUESTED** [review]
The skipping of the contig correction itself works well. However, when activating `--skip_ancient_damagecorrection` pyDamage doesn't export the files into a sample specific sub-folder any longer. The usual way was that pyDamage has sample-specific sub-folders in `Ancient_DNA/pydamage/analyze` and `Ancient_DNA/pydamage/filter`. At the moment, for every sample the same file is used and the results of the previous run sample are overwritten.

File: conf/modules.config [hunk 1]
```diff
@@ -521,7 +521,7 @@ process {
     withName: PYDAMAGE_ANALYZE {
         ext.prefix = { "${meta.assembler}-${meta.id}" }
         publishDir = [
-            path: {"${params.outdir}/Ancient_DNA/pydamage/analyze/" },
+            path: {"${params.outdir}/Ancient_DNA/pydamage/analyze/"${meta.assembler}-${meta.id}" },
```
Comment (jfy133) [inline conf/modules.config:524]:
```suggestion
            path: {"${params.outdir}/Ancient_DNA/pydamage/analyze/${meta.assembler}-${meta.id}" },
```

File: conf/modules.config [hunk 2]
```diff
@@ -530,7 +530,7 @@ process {
         ext.prefix = { "${meta.assembler}-${meta.id}" }
         ext.args   = "-t ${params.pydamage_accuracy}"
         publishDir = [
-            path: {"${params.outdir}/Ancient_DNA/pydamage/filter/" },
+            path: {"${params.outdir}/Ancient_DNA/pydamage/filter/"${meta.assembler}-${meta.id}" },
```
Comment (jfy133) [inline conf/modules.config:533]:
```suggestion
            path: {"${params.outdir}/Ancient_DNA/pydamage/filter/${meta.assembler}-${meta.id}" },
```

**Comment by jfy133** [comment]
Fix applied, waiting from results on tests as discussed on slack.

**Review by alexhbnr — APPROVED** [review]

**Pull request merged**